Restructure and reflow security section (#353)
quicwg/qlog: draft-ietf-quic-qlog-main-schema.md

# Security and privacy considerations {#privacy}

Protocols such as TLS {{?RFC8446}} and QUIC {{?RFC9000}} provide varying degrees
of secure protection for the wire image {{?RFC8546}}. There is inevitably
tension between security and observability, when logging can reveal aspects of

> the wire image, that would ordinarily be protected. This tension equally applies

Reviewer: ```suggestion
the wire image that would ordinarily be protected. This tension equally applies
```

to any privacy considerations that build on security properties, especially if
data can be correlated across data sources.

Any data that is determined to be necessary for a use case at hand could be

> logged or captured. As per {{?RFC6973}}, operators must be aware that such data

Reviewer: This first sentence is not precise about whether it refers to the logging use case or the use case of the logged exchange.

Author: I agree. I'm going to track this in a separate ticket (along with your other mention of use-cases)

  values, HTTP response data, TLS SNI field values), which can contain directly
  sensitive information.

## Operational implications and recommendations

> Operational implications qlog should focus on log capture and access.

Reviewer: This doesn't appear to be a complete and well-formed sentence.

It is recommended that the ability to enable qlog capture is subject to access
control and auditing. Furthermore, access control policy should extend to the
level of information that is allowed to be captured (e.g., capture of

> more-sensitive data requires higher privileges).

Reviewer: Would it be reasonable to suggest that any authorization for logging access be considered equivalent to plaintext access to communications.  I realize that some people might feel uncomfortable about that, because they enable logging where they would not enable plaintext access, but I think that considering them equivalent -- absent strong justification to the contrary -- better reflects the true nature of most communications.

Author: Yes, I'll update the text to reflect that.

It is recommended that access to stored qlogs is subject to access control and
auditing.

Access control techniques in end user client environments can be limited.

> End-users have the possibility of capturing their own logs and risking their own
> security and privacy in ways that could be surprising. Implementations should

Reviewer: This statement is a bit obtuse.  I would instead say:

```suggestion
An end-user that might enable logging without understanding the implications of
that choice on their privacy and security. Implementations should
```

consider how to make enabling qlog conspicous, and resistant to automation or

> drive-by attacks. Examples include, requiring explicit actions to start a

Reviewer: ... or social engineering?

Author: yep that was my implication but I'll add that

> capture, and isolation or sandboxing of capture from other activities in the

Reviewer: ```suggestion
capture and isolation or sandboxing of capture from other activities in the
```

same process or component.

It is recommended that data retention policies are defined for the storage of
qlog files.

It is recommended that qlog files are encrypted in transit and at rest.

## Data minimization or anonymization

Data minimization or anonymization techniques can be applied to qlog. While this
can go some way to addressing security and privacy risks, implementers and

> operators are advised that the techniques might not provide a sufficient level

Reviewer: ```suggestion
operators are advised that removing or anonymizing data without sufficient care might
not improve privacy or securrity.
```

> of mitigation. Equally, minimization or anonymization can reduce the
> effectiveness of qlog data.

Reviewer: ```suggestion
Minimization or anonymization might greatly reduce the usefulness
of qlog data.
```

It is rare that the effect will be equal.  (You probably meant "at the same time" or "in the same way")

> Operator and implementors need to balance logging fields against the potential

Reviewer: ```suggestion
Operator and implementers need to balance the value of logged data against the potential
```

risks inherent in their (involuntary) disclosure. This balance depends on the
use case at hand (e.g., research datasets might have different requirements to

> live operational troubleshooting).

Reviewer: It seems like the difference in these examples is WHO has access to the data and the level to which the different entities might be trusted, so maybe this isn't so much about use cases.
